src/provider/mod.rs

```diff
@@ -57,13 +58,20 @@ impl Server {
     }
 }
 
+#[derive(Debug)]
+pub struct ConfigDefault {
+    pub key: Config,
+    pub value: &'static str,
+}
+
 #[derive(Debug)]
 pub struct Provider {
     pub status: Status,
     pub before_login_hint: &'static str,
     pub after_login_hint: &'static str,
     pub overview_page: &'static str,
     pub server: Vec<Server>,
+    pub config_defaults: Option<Vec<ConfigDefault>>,
```

Comment:
The only thing that caught my eye was that you first have to look what a `ConfigDefault` is here. Maybe a `HashMap<Config, &'static str>` would be better, but I guess that it's not worth it to change it now.

Reply:
`config_defaults` is never accessed randomly and it is iterated only once. so i think, a HashMap is overhead here. also a the construction is more expensive - and lazy_config will construct everything even if we are interested in only one. but all this is theoretically, in practise we will maybe have a handful of these default-options. and, we can also change it easily as needed.